src/open_company_web/components/topic_list.cljs

```diff
@@ -179,7 +177,133 @@
      :fullscreen-force-edit (if (nil? current-state) (router/section-editing?) (:fullscreen-force-edit current-state))
      :show-add-topic-tip show-add-topic-tip
      :show-second-add-topic-tooltip (or (:show-second-add-topic-tooltip current-state) false)
-     :show-share-tooltip (or (:show-share-tooltip current-state) false)}))
+     :show-share-tooltip (or (:show-share-tooltip current-state) false)
+     :show-second-pin-tip (or (:show-second-pin-tip current-state) false)
+     :rerender (rand 4)}))
+
+(defn save-sections-order [owner]
+  (let [col1-pinned-topics (sel [:div.col-1 :div.topic.draggable-topic])
+        col1-pinned-topics-list (vec (for [topic col1-pinned-topics] (.data (js/$ topic) "section")))
+        col2-pinned-topics (sel [:div.col-2 :div.topic.draggable-topic])
+        col2-pinned-topics-list (vec (for [topic col2-pinned-topics] (.data (js/$ topic) "section")))
+        col3-pinned-topics (sel [:div.col-3 :div.topic.draggable-topic])
+        col3-pinned-topics-list (vec (for [topic col3-pinned-topics] (.data (js/$ topic) "section")))
+        max-count (max (count col1-pinned-topics-list) (count col2-pinned-topics-list) (count col3-pinned-topics-list))
+        pinned-topics-list (loop [topics []
+                                  idx 0]
+                             (if (<= (inc idx) max-count)
+                               (recur (vec (remove nil? (conj topics (get col1-pinned-topics-list idx) (get col2-pinned-topics-list idx) (get col3-pinned-topics-list idx))))
+                                      (inc idx))
+                               topics))
+        other-topics (sel [:div.topic.not-draggable-topic])
+        other-topics-list (vec (for [topic other-topics] (.data (js/$ topic) "section")))]
+    (dispatcher/dispatch! [:new-sections (vec (concat pinned-topics-list other-topics-list))])))
+
+(defn pinned-count [data]
```

Comment:
function docstring and possibly some inline comments would be good here. this function is non-trivial.